armi/physics/neutronics/settings.py

```diff
@@ -342,251 +343,170 @@ def defineSettings():
+                    CONF_GROUP_STRUCTURE, value, newValue
                 )
             )
-            return {name: cs.get("groupStructure").default}
-
-
-def _migrateDpa(_cs, name, value):
-    newValue = value
-    if value == "dpaHT9_33":
-        newValue = "dpaHT9_ANL33_TwrBol"
-    elif value == "dpa_SS316":
-        newValue = "dpaSS316_ANL33_TwrBol"
-
-    return {name: newValue}
-
-
-@include_as_rule("gridPlateDpaXsSet")
-def migrateGridPlateDpa(_cs, name, value):
-    """Got more rigorous in dpa XS names."""
-    return _migrateDpa(_cs, name, value)
-
 
-@include_as_rule("dpaXsSet")
-def migrateDpaXs(_cs, name, value):
-    return _migrateDpa(_cs, name, value)
-
-
-@include_as_rule("saveNeutronicsOutputs")
-def resetNeutronicsOutputsToSaveAfterRename(_cs, _name, value):
-    """
-    Reset the values of 'saveNeutronicsOutputs' setting after it was migrated to 'neutronicsOutputsToSave'.
-
-    Arguments
-    ---------
-    cs : setting object
-        ARMi object containing the default and user-specified settings.
-
-    name : str
-        Setting name to be modified by this rule.
-
-    value : str
-        Value of the setting identified by name.
-
-    Returns
-    -------
-    dict : dict
-        Updated setting name and or value.
-    """
-    newName = "neutronicsOutputsToSave"
-    if value == "True":
-        newValue = ALL
-    elif value == "False":
-        newValue = ""
+        inspector.cs = inspector.cs.modified(
+            newSettings={CONF_GROUP_STRUCTURE: newValue}
+        )
 
-    runLog.info(
-        "The `neutronicsOutputsToSave` setting has been set to {} based on deprecated "
-        "`saveNeutronicsOutputs`.".format(newValue)
+    queries.append(
+        settingsValidation.Query(
+            lambda: inspector.cs[CONF_GROUP_STRUCTURE] not in GROUP_STRUCTURE,
+            "The given group structure {0} was not recognized.".format(
+                inspector.cs[CONF_GROUP_STRUCTURE]
+            ),
+            "Would you like to auto-correct the group structure value?",
+            updateXSGroupStructure,
+        )
     )
-    return {newName: newValue}
-
-
-@include_as_rule("gammaTransportActive")
-def removeGammaTransportActive(cs, _name, value):
-    """
-    Remove 'gammaTransportActive' and set values of 'globalFluxActive' for the same functionality.
-
-    Arguments
-    ---------
-    cs : setting object
-        ARMi object containing the default and user-specified settings.
-
-    name : str
-        Setting name to be modified by this rule.
-
-    value : str
-        Value of the setting identified by name.
-    """
-    if value == "True":
-        newValue = NEUTRONGAMMA
-    elif value == "False":
-        newValue = NEUTRON
-
-    cs["globalFluxActive"] = newValue
-    runLog.info(
-        "The `globalFluxActive` setting has been set to {} based on deprecated "
-        "`gammaTransportActive`.".format(newValue)
+
+    def migrateDpa(name0):
```

Comment:
My feeling is that this belongs in our internal neutronics plugin, since we're making reference to TWR and specific materials here.

Reply:
I agree with that.

But we will have to copy/paste over the settingsValidator version, not the old settingsRules version.